runtime/calamari/tests/integration_tests.rs

```diff
@@ -0,0 +1,950 @@
+			let author = alice.clone();
+			let mut header = seal_header(
+				Header::new(
+					0,
+					Default::default(),
+					Default::default(),
+					Default::default(),
+					Default::default(),
+				),
+				author.clone(),
+			);
+
+			header.digest_mut().pop(); // pop the seal off.
+			calamari_runtime::System::initialize(&1, &Default::default(), header.digest());
+			assert_eq!(Authorship::author().unwrap(), author);
+
+			let call = Call::Balances(pallet_balances::Call::transfer {
+				dest: sp_runtime::MultiAddress::Id(charlie),
+				value: 10 * KMA,
+			});
+
+			let len = 10;
+			let info = info_from_weight(100);
+			let maybe_pre = ChargeTransactionPayment::<Runtime>::from(0)
+				.pre_dispatch(&bob, &call, &info, len)
+				.unwrap();
+
+			let res = call.clone().dispatch(Origin::signed(bob));
+
+			let post_info = match res {
+				Ok(info) => info,
+				Err(err) => err.post_info,
+			};
+
+			let _res = ChargeTransactionPayment::<Runtime>::post_dispatch(
+				Some(maybe_pre),
+				&info,
+				&post_info,
+				len,
+				&res.map(|_| ()).map_err(|e| e.error),
+			);
+
+			let author_received_reward = Balances::free_balance(alice) - INITIAL_BALANCE;
+			println!("The rewarded_amount is: {:?}", author_received_reward);
+
+			let author_percent = Percent::from_percent(FEES_PERCENTAGE_TO_AUTHOR);
+			let expected_fee =
+				TransactionPayment::compute_actual_fee(len as u32, &info, &post_info, 0);
+			assert_eq!(author_received_reward, author_percent * expected_fee);
+
+			let treasury_percent = Percent::from_percent(FEES_PERCENTAGE_TO_TREASURY);
+			assert_eq!(
+				Balances::free_balance(Treasury::account_id()),
+				treasury_percent * expected_fee
+			);
+		});
+}
+
+#[test]
+fn root_can_change_default_xcm_vers() {
+	ExtBuilder::default().build().execute_with(|| {
+		// Root sets the defaultXcm
+		assert_ok!(PolkadotXcm::force_default_xcm_version(
+			root_origin(),
+			Some(2)
+		));
+	})
+}
+
+#[test]
+fn sanity_check_session_period() {
+	assert_eq!(Period::get(), 6 * HOURS);
+}
+
+fn advance_session_assertions(session_index: &mut u32, advance_by: u32) {
+	*session_index += advance_by;
+
+	run_to_block(*session_index * Period::get() - 1);
+	assert_eq!(Session::session_index(), *session_index - 1);
+
+	run_to_block(*session_index * Period::get());
+	assert_eq!(Session::session_index(), *session_index);
+}
+
+#[test]
+fn session_and_collator_selection_work() {
+	let alice = get_account_id_from_seed::<sr25519::Public>("Alice");
+	let bob = get_account_id_from_seed::<sr25519::Public>("Bob");
+	let alice_aura = get_collator_keys_from_seed("Alice");
+	let bob_aura = get_collator_keys_from_seed("Bob");
+	let desired_candidates = 1;
+
+	ExtBuilder::default()
+		.with_collators(vec![alice.clone()], desired_candidates)
+		.with_balances(vec![
+			(alice.clone(), INITIAL_BALANCE),
+			(bob.clone(), INITIAL_BALANCE),
+		])
+		.build()
+		.execute_with(|| {
+			// Alice is in the invulnerables set, so not part of the candidates set.
```

Comment:
Just FYI, this assertion is only true for invulnerables at genesis, a candidate that is made invulnerable can be part of both sets.
(tbh. it's a bit strange to me that the latter is not always the case)